tokenclient/tokenFlows.go

```diff
@@ -128,18 +152,51 @@ func (t *TokenFlows) getURL(customerTenantURL string) (string, error) {
 	return "", fmt.Errorf("customer tenant url '%v' can't be parsed: %w", customerTenantURL, err)
 }
 
-func (t *TokenFlows) performRequest(r *http.Request, v interface{}) error {
-	res, err := t.options.HTTPClient.Do(r)
+func (t *TokenFlows) getOrRequestToken(r request) (string, error) {
+	// token cached?
+	cachedToken := t.readFromCache(&r)
+	if cachedToken != "" {
+		return cachedToken, nil
+	}
+
+	// request token
+	var tokenRes tokenResponse
+	err := t.performRequest(r, &tokenRes)
+	if err != nil {
+		return "", err
+	} else if tokenRes.Token == "" {
```

Comment:
Nit: Omitting the "else" is a bit more idiomatic'

Reply:
ok